src/librustc/mir/interpret/pointer.rs

```diff
@@ -202,6 +202,13 @@ impl<'tcx, Tag> Pointer<Tag> {
         Pointer { alloc_id: self.alloc_id, offset: self.offset, tag: () }
     }
 
+    /// Erase any allocation IDs.
+    ///
+    /// Used when comparing heap snapshots.
+    pub fn erase_id(self) -> Pointer<Tag, ()> {
```

Comment:
Please rename to `erase_alloc_id`.

But isn't it easier to just say `ptr1.offset == ptr2.offset`? Pointers with `AllocId` type `()` are kind of pointless (all puns intended).

Reply:
Oh, you are recursively doing through even for `Place`. But how can that make any sense, given that all AllocIds are considered equal at this point...?

Reply:
:+1: to `erase_alloc_id`. This is the part that makes `AllocId`s compare as equal. It means we can use the derived `PartialEq` impl. The alternative is to overload `AllocIdInvariant` for more types, which would also be fine.

Reply:
But why does it make any sense to compare just the offsets?

Reply:
Other code is responsible for comparing the allocation that each `AllocId` references, specifically the `PartialEq` implementation of `InterpRef`. Am I misunderstanding the question?